### PR TITLE
LORE-519 - Implement ArticleExporter API

### PR DESCRIPTION
**Pull request by jamieshepherd**
# Description
For our ML classification we want to be able to get plaintext version of article content, along with some associated metadata. Similar functionality is available in `WikiaSearchIndexer`, but for large pages it's especially slow. This is a fast, not-so-thorough API that exports only the fields we need for our ML classifier.

# Example
https://gameofthrones.jshepherd.fandom-dev.us/wikia.php?controller=ArticleExporterApi&method=getArticles&ids[]=201

@Wikia/lore

### CODE REVIEW

**Review by nmonterroso — APPROVED** [review]

**Comment by drsnyder** [comment]
Can we add some tests?

**Comment by drsnyder** [comment]
I'm getting a 404. @jamieshepherd did you branch change?

```
$ curl -s 'https://gameofthrones.jshepherd.fandom-dev.us/wikia.php?controller=ArticleExporterApi&method=getArticles&ids%5B%5D=201' | jq .
{
  "exception": {
    "type": "ControllerNotFoundException",
    "message": "Not found",
    "code": 404,
    "details": "Controller not found: ArticleExporterApi"
  },
  "trace_id": "0ebc1c83-8d7d-408d-aef3-bb1a23085652"
}
```

**Comment by jamieshepherd** [comment]
@drsnyder yep - I've re-upped lore-519. You can check that or checkout on your dev box.

File: extensions/wikia/ArticleExporter/ArticleExporterApiController.class.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		global $wgRequest;
+
+		$ids = $wgRequest->getArray( 'ids' );
+
+		$exporter = new ArticleExporter();
+        $articles = $exporter->build($wgCityId, $ids);
+
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+		$this->response->setValues( [ 'articles' => $articles ] );
+	}
```
Comment (drsnyder) [inline extensions/wikia/ArticleExporter/ArticleExporterApiController.class.php:21]: Looks like some of the spacing might be off.

Comment (jamieshepherd) [reply extensions/wikia/ArticleExporter/ArticleExporterApiController.class.php:21]: Fixed in b927b61

**Review by drsnyder — APPROVED** [review]

File: extensions/wikia/ArticleExporter/ArticleExporter.class.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                $articles[$id] = [
+                    'wikiId' => $cityId,
+                    'pageId' => $id,
+                    'revisionId' => strval( $article['parse']['revid'] ),
+                    'title' => $article['parse']['title'],
+                    'url' => $this->getPageUrl( $id ),
+                    'plaintextContent' => $this->getPlaintext( $article['parse']['text']['*'] ),
+                    'categories' => $this->getCategories( $article['parse']['categories'] ),
+                    'linkedPageTitles' => $this->getPageTitles( $article['parse']['links'] ),
+                    'updatedUtc' => $this->getUpdated( $article['parse']['revid'] ),
```
Comment (adaball) [inline extensions/wikia/ArticleExporter/ArticleExporter.class.php]: Sorry to be late to the party here, we can do it in another PR if needed. 

From a discussion [here](https://wikia.slack.com/archives/CJ9104MF0/p1557498292273000?thread_ts=1557443730.272200&cid=CJ9104MF0) it seems that we're going to need the language/locale.  Is it possible to update the returned payload to include this as well?

Comment (drsnyder) [inline extensions/wikia/ArticleExporter/ArticleExporter.class.php]: Can we also add the namespace id? That'll help the caller to potentially filter pages that we might want to ignore such as the talk pages.

Comment (nmonterroso) [reply extensions/wikia/ArticleExporter/ArticleExporter.class.php]: Added namespace in a12368d.

Comment (nmonterroso) [reply extensions/wikia/ArticleExporter/ArticleExporter.class.php]: 5ccacac9fd0
